### PR TITLE
add nwmatcher v1.3.6 with git auto-update, close #5433

### LINKED ISSUE #5433: Update nwmatcher
nwmatcher is currently on `v1.3.6` and the latest version on cdnjs is `1.2.5`. Would be awesome if you guys could update to the latest version. I'll try to find some time to make a PR :)

### PR DESCRIPTION
**Pull request by Amomo**
Hi @Piicksarn  , I just add git auto-update to update nwmatcher, from the request #5433
repo: https://github.com/dperini/nwmatcher
Would you mind checking this PR for me? Thank you~ :-)

- [ ] Not found on cdnjs repo
- [x] No already exist issue and PR
- [x] Notable popularity : **Watch `15`, Star `237`, Fork `24`**
- [x] Project has public repository on famous online hosting platform (or been hosted on npm) 
- [x] Has valid tags for each versions (for git auto-update)
- [x] Auto-update setup
- [x] Auto-update target/source is valid.
- [x] Auto-update filemap is correct.

### CODE REVIEW

**Comment by Piicksarn** [comment]
@Amomo, we had better follow the structure before.
The link below shown the files we need to include in the autoupdate config.
and according to the website, we add  `nwmatcher.js` only. 
https://cdnjs.com/libraries/nwmatcher

**Comment by Amomo** [comment]
@Piicksarn thanks for the suggestion!
I have modified it, could you take a look again? Thank you~

**Comment by Piicksarn** [comment]
@PeterDaveHello, would you please take a look at the pr?
Thank you!

File: ajax/libs/nwmatcher/package.json [hunk 1]
```diff
@@ -22,5 +22,13 @@
       "type": "git",
       "url": "git://github.com/dperini/nwmatcher.git"
     }
-  ]
+  ],
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/.git",
```
Comment (PeterDaveHello) [inline ajax/libs/nwmatcher/package.json]: ...

Comment (Piicksarn) [inline ajax/libs/nwmatcher/package.json]: ha! sorry!!
@Amomo, take a look!~

Comment (Amomo) [inline ajax/libs/nwmatcher/package.json]: Ohhhh! so embarrassing😂 I will fix it up.

**Comment by Amomo** [comment]
@Piicksarn I have corrected it!
Could you take a look again? Thank you~ ☺️

**Comment by Piicksarn** [comment]
@PeterDaveHello, it's fine now!
would you please have a look at the PR？
THANK YOU!

**Comment by PeterDaveHello** [comment]
Don't we need other files?

**Comment by Piicksarn** [comment]
@PeterDaveHello, we followed the structure before.
Since the version included in cdnjs, it haven't include the other files.
and the **Installation** in [README](https://github.com/dperini/nwmatcher/blob/master/README.md) shows that nwmatcher.js is main file!
So we just add nwmatcher.js in file map. :^D

**Comment by PeterDaveHello** [comment]
I think we can add `nwmatcher-base.js` & `nwmatcher-noqsa.js`.

**Comment by Amomo** [comment]
@Piicksarn I have modified it, could you take a look again? Thank you~

**Comment by Piicksarn** [comment]
@PeterDaveHello would you please take a look at the PR?
Thank you!

**Comment by PeterDaveHello** [comment]
Rebase first.

**Comment by Amomo** [comment]
@PeterDaveHello Ok, I have rebased it!

**Comment by Piicksarn** [comment]
@PeterDaveHello, @Amomo has it rebased!

File: ajax/libs/nwmatcher/package.json [hunk 2]
```diff
@@ -22,5 +22,17 @@
...
+  ],
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/dperini/nwmatcher.git",
+    "basePath": "src/",
+    "files": [
+      "nwmatcher*.js"
+    ]
+  },
+  "requiredFile": [
```
Comment (PeterDaveHello) [inline ajax/libs/nwmatcher/package.json]: Wrong field name  ...

Comment (Amomo) [inline ajax/libs/nwmatcher/package.json]: Oh!! Sorry for the misuse! 😱
Uh...what should I use for this field?
Could we write it down in the document?

**Comment by Amomo** [comment]
@Piicksarn I have modified it, could you take a look again? 
Many thanks 😭

**Comment by Piicksarn** [comment]
@PeterDaveHello would you please take a look at the pr?
thanks!

**Pull request merged**